### DISCUSSION TITLE
Reboot based on boottime

### DISCUSSION THREAD

**Discussion by nyr3188** [travisghansen/hass-pfsense#173] [Q&A]
I know that sensor.firewall_system_boottime returns a date and time. Can I trigger a reboot if the current date and time is greater than 30 days away from the boottime?

**Comment by sikysikov** [comment]
I would like reboot entity too.

Rebooting pfSense in specific time pattern can be done using Cron in pfSense.

**Comment by travisghansen** [comment]
`system_reboot` and `system_halt` are services you can invoke on any entity.

**Reply by sikysikov** [reply]
Hi, great, thank you.

**Reply by sikysikov** [reply]
Hi, calling service to reboot pfSense ends in script error, but pfSense rebooted:

Result:
params:
  domain: pfsense
  service: system_reboot
  service_data:
    entity_id: binary_sensor.pfsense_localdomain_pending_notices_present
  target: {}
running_script: false

**Reply by travisghansen** [reply]
If I recall correctly that’s somewhat expected as the webserver shuts down immediately. I would have to do some testing again but I wouldn’t worry about it too much.

**Reply by sikysikov** [reply]
Ok, but there is little issue that automation script stops due to this error.
